sql/src/main/java/io/crate/metadata/sys/SysRepositoriesTableInfo.java

```diff
@@ -51,15 +55,18 @@
         static final ColumnIdent SETTINGS = new ColumnIdent("settings");
     }
 
-    public static ImmutableMap<ColumnIdent, RowCollectExpressionFactory<Repository>> expressions() {
+    public static ImmutableMap<ColumnIdent, RowCollectExpressionFactory<Repository>> expressions(List<Setting<?>> maskedSettings) {
         return ImmutableMap.<ColumnIdent, RowCollectExpressionFactory<Repository>>builder()
             .put(SysRepositoriesTableInfo.Columns.NAME,
                 () -> NestableCollectExpression.forFunction((Repository r) -> r.getMetadata().name()))
             .put(SysRepositoriesTableInfo.Columns.TYPE,
                 () -> NestableCollectExpression.forFunction((Repository r) -> r.getMetadata().type()))
             .put(SysRepositoriesTableInfo.Columns.SETTINGS,
                 () -> NestableCollectExpression
-                    .forFunction((Repository r) -> r.getMetadata().settings().getAsStructuredMap()))
+                    .forFunction(r -> r.getMetadata().settings().getAsStructuredMap(maskedSettings
```

Comment:
Do we want to do this change now or when we convert sensitive settings to `masked settings`? Imho, it would be more logically complete change if we do it later.

Reply:
yeap, also saw this and I'm cleaning up now - thanks